controllers/snapshot/snapshot_adapter.go

```diff
@@ -79,6 +81,11 @@ func (a *Adapter) EnsureAllIntegrationTestPipelinesExist() (results.OperationRes
 			"IntegrationTestScenarios", len(*integrationTestScenarios))
 		for _, integrationTestScenario := range *integrationTestScenarios {
 			integrationTestScenario := integrationTestScenario //G601
+			if &integrationTestScenario.Spec.Environment != nil {
+				//get the environmet according to environment name from integrationTestScenario
+				a.logger.Info("IntegrationTestScenario has environment defined, skipping creation of pipelinerun.", "IntegrationTestScenario: ", integrationTestScenario)
```

Comment:
Could you explain why we skip pipelinerun when `&integrationTestScenario.Spec.Environment != nil`? Thanks!

Reply:
We don't want to create pipelineruns for integrationTestScenarios which have environment defined, it is also mentioned in story:
`Lastly, the Integration service needs to skip creating Integration test Tekton PipelineRuns for the IntegrationTestScenarios that require an environment.`

Reply:
The `pipelineRuns` for `IntegrationTestScenarios` that require an `Environment` will need to be created only after the `Environment` has been prepared. For that we will have a new controller as part of [HACBS-1233](https://issues.redhat.com/browse/HACBS-1233) that will start those `pipelineRuns` after the ephemeral Environment is ready for testing.

Reply:
I see and recall it now. Thank you so much Jiri, Kruno.